### PR TITLE
[Relay] Add fast_softmax

### PR DESCRIPTION
**Pull request by merrymercy**
There is an exponent operation in softmax. We can use `fast_exp` from #4790 to accelerate it.
This pr adds a new op `topi.nn.fast_softmax` to use this optimization.

I don't have a manual schedule for it, so this op should only be used together with auto-scheduler. I don't add it to relay's FastMath pass, because other people might use FastMath pass without auto-scheduler.

To use this op, users can create a new relay pass to do the op substitution.

### CODE REVIEW

**Comment by merrymercy** [comment]
cc @alexgl-github  @antinucleon

File: python/tvm/relay/op/nn/nn.py [hunk 1]
```diff
@@ -698,6 +698,29 @@ def softmax(data, axis=-1):
     return _make.softmax(data, axis)
 
 
+def fast_softmax(data, axis=-1):
+    r"""Computes softmax.
+    Use approximation to compute exponent for faster speed.
+
+    .. math:: \text{softmax}(x)_i = \frac{exp(x_i)}{\sum_j exp(x_j)}
```
Comment (junrushao1994) [inline python/tvm/relay/op/nn/nn.py:705]: would you like to update the formula to reflect we are using alternative ways to compute the softmax?

**Review by junrushao1994 — COMMENTED** [review]
The implementation looks good overall.

**Comment by junrushao1994** [comment]
The impl is good, I was just wondering if the fast softmax is biased or not?

File: python/tvm/relay/op/nn/nn.py (hunk 1 above)
Comment (merrymercy) [reply python/tvm/relay/op/nn/nn.py:705]: The math formula is still the same, but we use some approximation to compute exp.
The approximation cannot be described as short formulas, so we only mentioned this by text in the doc string at L703.

Comment (junrushao1994) [reply python/tvm/relay/op/nn/nn.py:705]: Okay I am not going to block this.

**Review by junrushao1994 — APPROVED** [review]

**Pull request merged**